addons/mail/static/src/core/common/message_reaction_list.js

```diff
@@ -0,0 +1,58 @@
+/* @odoo-module */
+
+import { Component } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+
+export class MessageReactionList extends Component {
+    static template = "mail.MessageReaction.List";
+    static props = {
+        id: { type: Number, required: true },
+        close: { type: Function, required: true },
+        reaction: { type: Object, required: true },
+        openReactionMenu: { type: Function, required: true },
+    };
+
+    getReactionSummary() {
+        const reaction = this.props.reaction;
+        const [firstUserName, secondUserName, thirdUserName] = reaction.personas.map(
+            ({ name, displayName }) => name || displayName
+        );
+        switch (reaction.count) {
+            case 1:
+                return _t("%s has reacted with %s", firstUserName, reaction.content);
+            case 2:
+                return _t(
+                    "%s and %s have reacted with %s",
+                    firstUserName,
+                    secondUserName,
+                    reaction.content
+                );
+            case 3:
+                return _t(
+                    "%s, %s, %s have reacted with %s",
+                    firstUserName,
+                    secondUserName,
+                    thirdUserName,
+                    reaction.content
+                );
+            default:
+                return {
+                    names: _t("%s, %s, %s and ", firstUserName, secondUserName, thirdUserName),
+                    highlightText: _t("%s other", reaction.personas.length - 3),
+                    content: _t(" person have reacted with %s", reaction.content),
+                };
+        }
```

Comment:
I think the format of string should be worded to show reaction before the persons:
`%(reaction)s reacted by %(person)s`

Reasoning is that the item at hand is the reaction, so it feels best to see item and name of reaction in title close together rather than seeing named reaction at the very end of the popover.
Also "other person" plural can simply be handled with "other(s)", which is more brief.